### PR TITLE
[USER] Added UserService

### PR DESCRIPTION
**Pull request by OleksandrSaven**

### CODE REVIEW

File: src/main/java/com/project/carsharingapp/model/Role.java [hunk 1]
```diff
@@ -22,4 +22,9 @@ public class Role {
     @Column(name = "name", nullable = false, unique = true)
     @Enumerated(EnumType.STRING)
     private RoleName roleName;
+
+    public enum RoleName {
+        MANAGER,
+        CUSTOMER
```
Comment (DmytroV95) [inline src/main/java/com/project/carsharingapp/model/Role.java]:
```suggestion
        ROLE_MANAGER,
        ROLE_CUSTOMER

```

File: src/main/java/com/project/carsharingapp/services/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    public UserDto findById(Long id) {
+        User user = userRepository.findUserById(id).orElseThrow(
+                () -> new RuntimeException("Can't find user by id " + id));
+        return userMapper.toDto(user);
+    }
+
+    @Override
+    public UserDto updateUserProfile(Long id, UpdateUserProfileRequestDto requestDto) {
+        User user = userRepository.findUserById(id).orElseThrow(
+                () -> new RuntimeException("Can't find user by id " + id));
```
Comment (DmytroV95) [inline src/main/java/com/project/carsharingapp/services/impl/UserServiceImpl.java]: You are using this code in different methods. Make it as separate method and reuse it. Also use our custom EntityNotFoundException

File: src/main/java/com/project/carsharingapp/services/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        user.setFirstName(requestDto.getFirstName());
+        user.setLastName(requestDto.getLastName());
+        user.setPassword(requestDto.getPassword());
+        return userMapper.toDto(userRepository.save(user));
+    }
+
+    @Override
+    public UserDto updateUserRole(Long id, UpdateUserRoleRequestDto requestDto) {
+        User user = userRepository.findUserById(id).orElseThrow(
+                () -> new RuntimeException("Can't find user by id " + id));
```
Comment (DmytroV95) [inline src/main/java/com/project/carsharingapp/services/impl/UserServiceImpl.java]: Same

File: src/main/java/com/project/carsharingapp/services/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    private final RoleRepository roleRepository;
+
+    @Override
+    public UserDto findById(Long id) {
+        User user = userRepository.findUserById(id).orElseThrow(
+                () -> new RuntimeException("Can't find user by id " + id));
```
Comment (DmytroV95) [inline src/main/java/com/project/carsharingapp/services/impl/UserServiceImpl.java]: Same

**Review by DmytroV95 — CHANGES_REQUESTED** [review]
GJ! Left comments to improve your solution.

File: src/main/java/com/project/carsharingapp/services/impl/UserServiceImpl.java (hunk 2 above)
Comment (OleksandrSaven) [reply src/main/java/com/project/carsharingapp/services/impl/UserServiceImpl.java]: I know, when I made this  service, there was no GlobalExeptionHandler

**Review by DmytroV95 — APPROVED** [review]
LGTM!

**Pull request merged**